hail/src/main/scala/is/hail/expr/ir/functions/Functions.scala

```diff
@@ -82,6 +83,39 @@ object IRFunctionRegistry {
       })
   }
 
+  def pyRegisterIRForServiceBackend(
+    ctx: ExecuteContext,
+    name: String,
+    typeParamStrs: Array[String],
+    argNames: Array[String],
+    argTypeStrs: Array[String],
+    returnType: String,
+    bodyStr: String
+  ): Unit = {
+    requireJavaIdentifier(name)
+
+    val typeParameters = typeParamStrs.map(IRParser.parseType).toFastIndexedSeq
+    val valueParameterTypes = argTypeStrs.map(IRParser.parseType).toFastIndexedSeq
+    val refMap = argNames.zip(valueParameterTypes).toMap
+    val body = IRParser.parse_value_ir(
+      bodyStr,
+      IRParserEnvironment(ctx, refMap, Map())
+    )
+
+    userAddedFunctions += ((name, (body.typ, typeParameters, valueParameterTypes)))
+    addIR(
+      name,
+      typeParameters,
```

Comment:
Can someone on compilers team check this block?

Reply:
@johnc1231  can you take a look? My intent is for this to be exactly the same as above except `argNames` is now an `Array` instead of a `java.utils.ArrayList`.

Reply:
@johnc1231 bump

Reply:
I'm fine with this :+1: